lisp/mastodon-http.el

```diff
@@ -83,10 +84,17 @@ Authorization header is included by default unless UNAUTHENTICED-P is non-nil."
         (url-request-extra-headers
 	 (append
 	  (unless unauthenticed-p
-	    `(("Authorization" . ,(concat "Bearer " (mastodon-auth--access-token)))))
+	    `(("Authorization" . ,(concat "Bearer " (if (string= "oauth2" mastodon-auth-mechanism)
```

Comment:
Is this header needed at all for oauth2 or should the `(string= "oauth2" mastodon-auth-mechanism)` check go into the `unless` above to totally bypass this in the new flow?

Reply:
I have updated post and get both.